Game Logic/Game.cs

```diff
@@ -8,18 +8,128 @@ namespace Game_Logic
 {
     public class Game
     {
-        private GameBoard m_board;
+        private GameBoard m_Board;
         private Player m_PlayerO;
         private Player m_PlayerX;
 
-        public Game(int i_BoardSize)
+        public Game(int i_BoardSize, int i_NumOfPlayers, string i_PlayerName1, string i_PlayerName2 = "computer")
         {
-            m_board = new GameBoard(i_BoardSize);
+            m_Board = new GameBoard(i_BoardSize);
+            m_PlayerO = new Player(i_PlayerName1, eToken.O, ePlayerType.Human);
+            m_PlayerX = new Player(i_PlayerName2, eToken.X);
+
+            if(i_NumOfPlayers == 2)
+            {
+                m_PlayerX.PlayerType = ePlayerType.Human;
+            }
+
+            initializePiecesForPlayers();
+        }
+
+        private void initializePiecesForPlayers()
+        {
+            Cell[,] board = m_Board.getBoard();
+
         }
+
         public void PrintBoard()
         {
-            m_board.PrintBoard();
+            m_Board.PrintBoard();
+        }
+
+        public void UpdateValidMoves()
+        {
+
+        }
+
+        private void generateValidMovesForPiece(Piece i_Piece)
```

Comment:
/* To consider:

1. Adding an enum for directions to move
2. Using the Piece type with the directions enum and make the method more readable